presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -365,37 +358,70 @@ private QuerySpecification getRewrittenQuerySpecification(Metadata metadata, Qua
         Table materializedViewTable = new Table(QualifiedName.of(materializedView.getTable()));
         Query materializedViewQuery = (Query) sqlParser.createStatement(materializedView.getOriginalSql(), createParsingOptions(session));
 
-        return new QuerySpecificationRewriter(materializedViewTable, materializedViewQuery).rewrite(originalQuerySpecification);
+        return new QuerySpecificationRewriter(materializedViewTable, materializedViewQuery, materializedViewName).rewrite(originalQuerySpecification);
     }
 
     private class QuerySpecificationRewriter
             extends AstVisitor<Node, Void>
     {
-        private final Table materializedView;
+        Logger log = Logger.get(QuerySpecificationRewriter.class);
+        private final Table materializedViewTable;
         private final Query materializedViewQuery;
+        private final QualifiedObjectName materializedViewName;
 
         private MaterializedViewInfo materializedViewInfo;
         private Optional<Identifier> removablePrefix = Optional.empty();
         private Optional<Set<Expression>> expressionsInGroupBy = Optional.empty();
+        private Optional<String> errorMessage = Optional.empty();
 
         QuerySpecificationRewriter(
-                Table materializedView,
-                Query materializedViewQuery)
+                Table materializedViewTable,
+                Query materializedViewQuery,
+                QualifiedObjectName materializedViewName)
         {
-            this.materializedView = requireNonNull(materializedView, "materialized view is null");
+            this.materializedViewTable = requireNonNull(materializedViewTable, "materialized view is null");
             this.materializedViewQuery = requireNonNull(materializedViewQuery, "materialized view query is null");
+            this.materializedViewName = requireNonNull(materializedViewName, "materialized view name is null");
         }
 
         public QuerySpecification rewrite(QuerySpecification querySpecification)
         {
+            MaterializedViewInformationExtractor materializedViewInformationExtractor = new MaterializedViewInformationExtractor();
+            materializedViewInformationExtractor.process(materializedViewQuery);
+            materializedViewInfo = materializedViewInformationExtractor.getMaterializedViewInfo();
+            if (materializedViewInfo.getErrorMessage().isPresent()) {
+                log.debug("Failed to rewrite query with materialized view with following exception: %s", materializedViewInfo.getErrorMessage().get());
```

Comment:
Log it as a warning or error, debug messages are not logged by default.

Reply:
This was intentional- I didn't want to be logging due to non-exceptional failures.